jax/lax/lax.py

```diff
@@ -681,6 +682,10 @@ def reshape(operand: Array, new_sizes: Shape,
   """Wraps XLA's `Reshape
   <https://www.tensorflow.org/xla/operation_semantics#reshape>`_
   operator.
+
+  For inserting/removing dimensions of size 1, prefer using ``lax.squeeze`` /
+  ``lax.expand_dims``. These preserve information about axis identity that may
+  be useful for advanced transformation rules.
```

Comment:
I'm a little conflicted about whether it's worth adding this to the docstring or not. Might be better just to keep it as a code comment.